From 9260496c49bd5f1d299505cfadb8ca65fd8cff9c Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Wed, 10 Jul 2019 11:03:57 +0800
Subject: [PATCH 5/8] stb_sprintf.h: fix pointer detection

---
 stb_sprintf.h | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/stb_sprintf.h b/stb_sprintf.h
index 0635360..e124cc9 100644
--- a/stb_sprintf.h
+++ b/stb_sprintf.h
@@ -215,11 +215,8 @@ STBSP__PUBLICDEF void STB_SPRINTF_DECORATE(set_separators)(char comma, char peri
 #define stbsp__uint16 unsigned short
 
 #ifndef stbsp__uintptr
-#if defined(__ppc64__) || defined(__powerpc64__) || defined(__aarch64__) || defined(_M_X64) || defined(__x86_64__) || defined(__x86_64)
-#define stbsp__uintptr stbsp__uint64
-#else
-#define stbsp__uintptr stbsp__uint32
-#endif
+#include <stdint.h>
+#define stbsp__uintptr uintptr_t
 #endif
 
 #ifndef STB_SPRINTF_MSVC_MODE // used for MSVC2013 and earlier (MSVC2015 matches GCC)
-- 
2.30.0

